sites/main-site/src/content/blog/2024/maintainers-minutes-2024-10-25.mdx

```diff
@@ -0,0 +1,147 @@
+
+:::info
+This system only applies to modules nf-test, and should not be used at pipeline level!
+:::
+
+## Stalled PRs Due to Pending Reviews
+
+It has been noted that we occasionally receive almost-complete module PRs that stall because a reviewer leaves comments but does not return for a re-review.
+Even if the module author receives a second approving review, they often (since we’re considerate here at nf-core) feel uncomfortable merging without the original reviewer’s approval.
+
+After a brief discussion, we agreed to introduce a new reviewing guideline: if all reasonable comments have been addressed and another reviewer has approved, the PR should be merged after 1 month if the original reviewer has not returned for a re-review.
+
+If the original reviewer was temporarily unavailable during this period (which can happen!), the module can still be updated in a follow-up review.
+
+## Replacing monolithic `conf/modules.conf` with per-module configs
+
+A contentious point was brought up by the developers of [nf-core/rnaseq](https://nf-co.re/rnaseq) and [nf-core/methylseq](https://nf-co.re/methylseq).
+
+Both pipelines recently have experimented with modifying the ways in which modules and processes can get customised by config files.
+Currently, the nf-core template has a single file - `conf/modules.conf` - where all module-level configurations occur (specifying additional arguments, file publication specifications etc.).
+
+The two pipelines, however, have opted to split this single monolithic file into multiple files, with one config file per module e.g. [in this nf-core/methylseq PR](https://github.com/nf-core/methylseq/commit/7b276c97589b65153ae23b34fcf4f6da86bb8d70).
+
+The main goal of this approach is to make CI testing more efficient by allowing changes to a specific config to trigger only the related tests rather than the entire test suite (since all modules are currently dependent on a single config file).
+Another reason is to increase the modularity and portability of modules and their configurations.
+
+This proposal initially brought up a lot of consternation with most of the other maintainers present. The primary issue was the potential impact on usability, specifically regarding the ease with which users can locate customized parameters within each module.
+Currently, we can direct users to a single file within a single directory, simplifying their search.
+
+Splitting configurations into potentially hundreds of files could make it significantly harder for users to find what they need. Additionally, this approach isn’t currently supported by Nextflow’s native capabilities, which would allow modules to automatically detect and load nextflow.config files alongside main.nf. As a result, a substantial amount of manual work would be needed to manually include each module’s config.
```

Comment:
```suggestion
Splitting configurations into potentially hundreds of files could make it significantly harder for users to find what they need. It's also unclear if another file has an overriding setting for a module, for example that's defined in the subworkflow folder rather than the module folder. Additionally, this approach isn’t currently supported by Nextflow’s native capabilities, which would allow modules to automatically detect and load nextflow.config files alongside main.nf. As a result, a substantial amount of manual work would be needed to manually include each module’s config.
```